pkg/cmd/cli/cmd/loginoptions.go

```diff
@@ -92,17 +98,82 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 		}
 	}
 
-	// normalize the provided server to a format expected by config
-	serverNormalized, err := config.NormalizeServerURL(o.Server)
-	if err != nil {
+	// completeClientConfig checks if the provided host is reachable and if so, completes the clientConfig with
+	// CA data and other information required to make connections to the given server
+	if err := o.completeClientConfig(clientConfig); err != nil {
 		return nil, err
 	}
-	o.Server = serverNormalized
+	o.Config = clientConfig
+
+	return o.Config, nil
+}
+
+func (o *LoginOptions) completeClientConfig(clientConfig *restclient.Config) error {
+	if err := o.normalizeAndDialToServer(); err != nil {
+		clientConfig.Host = o.Server
+
+		switch {
+		case o.InsecureTLS:
+			clientConfig.Insecure = true
+			// insecure, clear CA info
+			clientConfig.CAFile = ""
+			clientConfig.CAData = nil
+		default:
+			switch err.(type) {
+			// certificate authority unknown, prompt user for insecure connection
+			case x509.UnknownAuthorityError:
+				// check to see if we already have a cluster stanza that tells us to use --insecure for this particular server.  If we don't, then prompt
+				clientConfigToTest := *clientConfig
+				clientConfigToTest.Insecure = true
+				matchingClusters := getMatchingClusters(clientConfigToTest, *o.StartingKubeConfig)
+
+				if len(matchingClusters) > 0 {
+					clientConfig.Insecure = true
+
+				} else if term.IsTerminal(o.Reader) {
+					fmt.Fprintln(o.Out, "The server uses a certificate signed by an unknown authority.")
+					fmt.Fprintln(o.Out, "You can bypass the certificate check, but any data you send to the server could be intercepted by others.")
+
+					clientConfig.Insecure = cmdutil.PromptForBool(os.Stdin, o.Out, "Use insecure connections? (y/n): ")
+					if !clientConfig.Insecure {
+						return fmt.Errorf(clientcmd.GetPrettyMessageFor(err))
+					}
+					// insecure, clear CA info
+					clientConfig.CAFile = ""
+					clientConfig.CAData = nil
+					fmt.Fprintln(o.Out)
+				}
+			// TLS record header errors, like oversized record which usually means the server only supports "http"
+			case tls.RecordHeaderError:
+				return fmt.Errorf(clientcmd.GetPrettyMessageFor(err))
+			default:
+				// try alternative ports before error out
+				if parsedURL, err := url.Parse(o.Server); err == nil {
+					if parsedHost, parsedPort, err := net.SplitHostPort(parsedURL.Host); err == nil {
+						alternativePortFor := map[string]string{"443": "8443"}
```

Comment:
You are right. I changed the approach here, we're only going to suggest the port in `defaultClusterURL` in case that's not what you're already using. But not submitting a request based on assumptions.